### ISSUE TITLE
[BUG] waagent is flodding serial port

### ISSUE THREAD

**Issue by ishmaelen** [Azure/WALinuxAgent#1571]
Hello

I see waagent flooding irq4 serial with more than 115200 bauds for some reason.

I do not know exactly what causes it as I do not trigger anything through the portal and I don't have a serial sniffer installed. Installing a VM using the latest Debian Stretch image w/ backports will give you the issue below. Disabling the agent will cause the messages in dmesg to disappear.

    VM Size: Standard D2s_v3

    # uname -a
    Linux intouch-hap02 4.19.0-0.bpo.5-cloud-amd64 #1 SMP Debian 4.19.37-3~bpo9+1 (2019-05-18) x86_64 GNU/Linux

    # waagent version
    WALinuxAgent-2.2.34 running on debian 9.9
    Python: 3.5.3
    Goal state agent: 2.2.34

    [Jul 1 07:43] serial8250: too much work for irq4
    [  +0.478310] serial8250: too much work for irq4
    [  +0.507968] serial8250: too much work for irq4
    [  +1.851085] serial8250: too much work for irq4
    [  +0.629418] serial8250: too much work for irq4

**Comment by vrdmr** [comment]
@ishmaelen - Can you please upload the walinuxagent logs (/var/log/waagent.log)? We'll need to investigate what are we flooding into the serial port.

**Comment by ishmaelen** [comment]
> @ishmaelen - Can you please upload the walinuxagent logs (/var/log/waagent.log)? We'll need to investigate what are we flooding into the serial port.

Thanks for investigating. Here is the log you are after : https://pastebin.com/1KNmGktg

**Comment by ishmaelen** [comment]
Same issue appearing in Linux 4.9.0-11-amd64 #1 SMP Debian 4.9.189-3+deb9u1 (2019-09-20) x86_64 GNU/Linux

**Comment by vrdmr** [comment]
Hi @ishmaelen, Apologies for the delay.

Can you please let me know the extensions installed on the box?
If you list `/var/lib/waagent`, it'll help me see whats is present on the box.

**Comment by vrdmr** [comment]
Thanks for the logs @ishmaelen and apologies for the delay. Based on the logs and the extensions installed on the box, I couldn't find any particular reason.

Can you please change the config `Logs.Console=n` in _/etc/waagent.conf_ (or add, if it is not present) and restart the agent? and check the serial port and see if the issue gets resolved?

Restart waagent service, for most of linux distributions:
```   
sudo service waagent restart
```
For Ubuntu, use:
```    
sudo service walinuxagent restart
```
For CoreOS, use:
```
sudo systemctl restart waagent
```

**Comment by DeterminedExecution** [comment]
I was able to correlate these messages in the ring buffer with the waagent.log by converting the ring buffer's seconds to dates and comparing those dates histogram of waagent.log message counts per minute.  

After compensating for local time vs UTC and a one hour DST offset I've determined the kernel error from too much serial console volume happens when waagent fetches a new goal state. 

Here is what the waagent.log shows during that time:

2020-06-14T21:42:01.674331Z INFO ExtHandler ExtHandler Fetching new goal state [incarnation 219 (new incarnation)]
2020-06-14T21:42:03.106646Z INFO ExtHandler ExtHandler Found private key matching thumbprint XXXXXXXXXXX
2020-06-14T21:42:03.126560Z INFO ExtHandler ExtHandler Fetch goal state completed
2020-06-14T21:42:03.210552Z INFO ExtHandler [Microsoft.OSTCExtensions.VMAccessForLinux-1.5.6] Target handler state: enabled [incarnation 219]
2020-06-14T21:42:03.217016Z INFO ExtHandler [Microsoft.OSTCExtensions.VMAccessForLinux-1.5.6] [Enable] current handler state is: enabled
2020-06-14T21:42:03.217899Z INFO ExtHandler [Microsoft.OSTCExtensions.VMAccessForLinux-1.5.6] Update settings file: 2.settings
2020-06-14T21:42:03.218968Z INFO ExtHandler [Microsoft.OSTCExtensions.VMAccessForLinux-1.5.6] Enable extension [extension_shim.sh -c ./vmaccess.py  -e]
2020-06-14T21:42:03.238521Z INFO ExtHandler ExtHandler Started extension using scope 'Microsoft.OSTCExtensions.VMAccessForLinux_1.5.6_XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX'
2020-06-14T21:42:03.251619Z INFO ExtHandler ExtHandler Started tracking new cgroup: Microsoft.OSTCExtensions.VMAccessForLinux-1.5.6, path: /sys/fs/cgroup/cpu/system.slice/Microsoft.OSTCExtensions.VMAccessForLinux_1.5.6XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX.scope
2020-06-14T21:42:03.262515Z INFO ExtHandler ExtHandler Started tracking new cgroup: Microsoft.OSTCExtensions.VMAccessForLinux-1.5.6, path: /sys/fs/cgroup/memory/system.slice/Microsoft.OSTCExtensions.VMAccessForLinux_1.5.6_XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX.scope
2020-06-14T21:42:06.320839Z INFO ExtHandler [Microsoft.Azure.RecoveryServices.VMSnapshotLinux-1.0.9163.0] Target handler state: enabled [incarnation 219]
2020-06-14T21:42:06.322675Z INFO ExtHandler [Microsoft.Azure.RecoveryServices.VMSnapshotLinux-1.0.9163.0] [Enable] current handler state is: enabled
2020-06-14T21:42:06.342240Z INFO ExtHandler [Microsoft.Azure.RecoveryServices.VMSnapshotLinux-1.0.9163.0] Update settings file: 217.settings
2020-06-14T21:42:06.351547Z INFO ExtHandler [Microsoft.Azure.RecoveryServices.VMSnapshotLinux-1.0.9163.0] Enable extension [main/handle.sh enable]
2020-06-14T21:42:06.379566Z INFO ExtHandler ExtHandler Started extension using scope 'Microsoft.Azure.RecoveryServices.VMSnapshotLinux_1.0.9163.0_XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX'
2020-06-14T21:42:06.380654Z INFO ExtHandler ExtHandler Started tracking new cgroup: Microsoft.Azure.RecoveryServices.VMSnapshotLinux-1.0.9163.0, path: /sys/fs/cgroup/cpu/system.slice/Microsoft.Azure.RecoveryServices.VMSnapshotLinux_1.0.9163.0_XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX.scope
2020-06-14T21:42:06.381596Z INFO ExtHandler ExtHandler Started tracking new cgroup: Microsoft.Azure.RecoveryServices.VMSnapshotLinux-1.0.9163.0, path: /sys/fs/cgroup/memory/system.slice/Microsoft.Azure.RecoveryServices.VMSnapshotLinux_1.0.9163.0_XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX.scope
2020-06-14T21:42:09.430259Z INFO ExtHandler ExtHandler ProcessGoalState completed [incarnation 219; 6294 ms]

**Comment by narrieta** [comment]
fixed by #2028 #2043
